pkg/controller/integration/integration_controller.go

```diff
@@ -534,6 +536,8 @@ func (r *reconcileIntegration) Reconcile(ctx context.Context, request reconcile.
 		}
 	}
 
+	metrics.UpdateIntegrationPhase(instance.Name, string(instance.Status.Phase))
```

Comment:
I think this would fit instead in https://github.com/apache/camel-k/blob/7ba6c99754f3fb826003df4e1f922890bca224ca/pkg/controller/integration/integration_controller.go#L88 as we already do for the readiness metric.